lib/email_repair/mechanic.rb

```diff
@@ -11,6 +11,26 @@ def repairs
       ]
     end
 
+    def repair_all(emails)
+      sanitized_emails = []
+      invalid_emails = []
+
+      emails.each do |email|
+        repaired_email = repair(email)
+
+        if repaired_email
+          sanitized_emails << repaired_email
+        else
+          invalid_emails << email
+        end
+      end
```

Comment:
Yeah, I was thinking I'd actually like the `repair` method to behave differently, as it seems like the current implementation makes for awkward code elsewhere.  Something like this might help:

```
def repair_all(emails)
  results = emails.map { |email| repair(email) }
  invalid_results, valid_results = results.partition(&:valid?)
 ... etc
end

def repair(email)
  ...
  OpenStruct.new(
    email: email, 
    sanitized_email: sanitized_email,
    valid?: sanitized_email.present?, 
  )
end
```